### PR TITLE
Adaptions to azure china infrastructure and documentation

### PR DESCRIPTION
**Pull request by mholkenbrink**
- Added a gpu nodepool to the azure infrastructure
- Renamed config_template.tfvars to terraform.tfvars.example (see #11) 
- Executed terraform fmt
- Adapted documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,11 +87,11 @@ To do so, please make a copy of the file `state-backend-template`, name it `stat
 
 ## Configuration
 
-For your configuration, please make a copy of the file `config_template.tfvars`, name it `config.tfvars` and open the file in a text editor. This file contains all variables that are configurable including documentation of the variables. Please adapt the values before you deploy the resources.
+For your configuration, please rename the file `terraform.tfvars.example` to `terraform.tfvars` and open the file in a text editor. This file contains all variables that are configurable including documentation of the variables. Please adapt the values before you deploy the resources.
```
Comment (mytracks) [inline README.md:90]: Why did you change this logic from "copying" to "rename"? I think if you rename it and make another "git pull" will the file come back?

File: k8s.tf [hunk 2]
```diff
@@ -125,12 +133,53 @@ resource "azurerm_kubernetes_cluster_node_pool" "execution-nodes" {
...
+  min_count             = var.gpuNodeCountMin
+  max_count             = var.gpuNodeCountMax
+  node_count            = var.gpuNodeCountMin
+  vm_size               = var.gpuNodeSize
+  max_pods              = 50
+  enable_auto_scaling   = true
+  vnet_subnet_id        = azurerm_subnet.gpu-nodes-subnet.id
+
+  node_labels = {
+    "purpose" = "execution" # Todo: clarify if Aurelion exec pods expect specific node labels
```
Comment (mytracks) [inline k8s.tf:162]: From my point of view the "purpose" must not be "execution" but "gpu". For the taints "purpose=gpu" is also used.

File: variables.tf [hunk 3]
```diff
@@ -1,78 +1,102 @@
...
+  type        = map(any)
   description = "The tags to be added to all resources."
   default     = {}
 }
 
-variable linuxNodeSize {
+variable "linuxNodeSize" {
   type        = string
   description = "The machine size of the Linux nodes for the regular services"
   default     = "Standard_D4s_v3"
```
Comment (mytracks) [inline variables.tf:30]: Currently, we already use "v4". However, there is already "v5" available.

File: variables.tf [hunk 4]
```diff
@@ -1,78 +1,102 @@
...
   type        = number
   description = "The maximum number of Linux nodes for the regular services"
   default     = 12
 }
 
-variable linuxExecutionNodeSize {
+variable "linuxExecutionNodeSize" {
   type        = string
   description = "The machine size of the Linux nodes for the job execution"
   default     = "Standard_D16s_v3"
```
Comment (mytracks) [inline variables.tf:48]: Currently, we already use "v4". However, there is already "v5" available.

File: variables.tf (hunk 3 above)
Comment (mholkenbrink) [reply variables.tf]: Fixed to v4 in 5669c7b18fa3e61302870da1d8caf3aa42de5316

File: k8s.tf (hunk 2 above)
Comment (mholkenbrink) [reply k8s.tf]: Fixed in 5669c7b18fa3e61302870da1d8caf3aa42de5316
